mathesar_ui/src/api/tables/columns.ts

```diff
@@ -1,4 +1,5 @@
 import type { DbType } from '@mathesar/AppTypes';
+import type { DurationUnit } from '@mathesar/utils/duration/types';
```

Comment:
I'd prefer to keep our application types dependent on the API types, and not the other way around. Can you move `DurationUnit` into the `src/api` somewhere? I recognize that it's not exactly relevant to `columns`, so we could potentially create a new directory like `src/api/_common` or something similar. I don't really care _where_ it goes within `src/api`, but I would like to put it in there somewhere.

It bears mentioning that I'd ideally like for `DbType` to live inside `src/api`, too -- but that type already existed when I created `src/api` and I wanted to minimize my refactoring at the time. (I'm not requesting you refactor `DbType` too -- only `DurationUnit` because that's new code.)

The fact that `DurationUnit` is a display option type does muddy the water somewhat, because (in theory) the front end should have control over display options. But even so, I find storing the type in the API directory to be helpful because it serves to communicate that any refactoring to the type requires extra care and thought beyond the references to that type within the rest of the font-end codebase. For example, if we refactor this type in a backwards incompatible way, we could still have runtime errors which slip through the type-checker due to user data that has not been migrated.

Reply:
@seancolsen I've made the changes in this commit: https://github.com/centerofci/mathesar/pull/1350/commits/6cac69c84e45f1486274a6bfbcdc817fc39acf88

I've placed it within `columns.ts` for the time-being because it also contains types like `NumberFormat` which aren't exactly relevant to `Columns`, same as `DurationUnit`. We can think of refactoring and moving these to a separate file later.